### DISCUSSION TITLE
liboqs integration with libcrypto.a?

### DISCUSSION THREAD

**Discussion by baluduvvuri1** [open-quantum-safe/liboqs#2005] [General]
Hi All,

I understand we can integrate liboqs to use OpenSSL (libcrypto.so) for crypto functionality by setting  OQS_USE_OPENSSL to ON during make.

**My query is can liboqs be integrated with statically build OpenSSL i.e libcrypto.a?**

For .so integration from the document:
https://github.com/open-quantum-safe/liboqs/blob/main/CONFIGURE.md#OQS_USE_OPENSSL
When OQS_USE_OPENSSL is ON, CMake also scans the filesystem to find the minimum version of OpenSSL required by liboqs (which happens to be 1.1.1). The [OPENSSL_ROOT_DIR](https://cmake.org/cmake/help/latest/module/FindOpenSSL.html) option can be set to aid CMake in its search.

Thank you

**Comment by SWilson4** [comment]
To the best of my knowledge, CMake will search for both shared and static libraries. I don't believe that we have an explicit test for building against libcrypto.a, but I don't see any reason why it shouldn't work. Are you encountering difficulties related to this?

**Comment by baluduvvuri1** [comment]
I am able to successfully search for the static library during compilation and then indicate both the libraries during build with final application.
Thank you for reverting

**Reply by MewanCdot** [reply]
Hi, what were the flags you passed to search for the static library?

**Reply by baluduvvuri1** [reply]
i have built openssl and static library libcrypto.a is present in same directory.

ls openssl-3.4.0-beta1/libcrypto.a
openssl-3.4.0-beta1/libcrypto.a

For liboqs to search for OpenSSL static library, I pass to cmake the following flags:

-DOQS_USE_OPENSSL=ON  
-DOPENSSL_INCLUDE_DIR=openssl-3.4.0-beta1/include 
-DOPENSSL_ROOT_DIR=openssl-3.4.0-beta1 

When cmake is run with the above flags, you should then see cmake log message like the below:

....
-- Found OpenSSL: openssl-3.4.0-beta1/libcrypto.a (Required is at least version "1.1.1")  
..

Let me know if you need additional help!
